api/types/database.go

```diff
@@ -565,13 +566,32 @@ func (d *DatabaseV3) setStaticFields() {
 	d.Version = V3
 }
 
+// validDatabaseNameRegexp filters the allowed characters in database names.
+// This is the (almost) the same regexp used to check for valid DNS 1035 labels,
+// except we allow uppercase chars.
+var validDatabaseNameRegexp = regexp.MustCompile(`^[a-zA-Z]([-a-zA-Z0-9]*[a-zA-Z0-9])?$`)
+
+// ValidateDatabaseName returns an error if a given string is not a valid
+// Database name.
+// Unlike application access proxy, database name doesn't necessarily
+// need to be a valid subdomain but use the same validation logic for the
+// simplicity and consistency, except two differences: don't restrict names to
+// 63 chars in length and allow upper case chars.
```

Comment:
> don't restrict names to 63 chars in length and allow upper case chars.

nit: maybe we can add a positive test case(s) with upper case chars and more than 63 chars to "guard" this

Reply:
added [261fd0e](https://github.com/gravitational/teleport/pull/28841/commits/261fd0e29bba0f9e61ce5d0662dc15a9f67e78e3)